cli/src/pcluster/validators/scheduler_plugin_validators.py

```diff
@@ -22,3 +23,67 @@ def _validate(self, grant_sudo_privileges: bool, requires_sudo_privileges: bool)
                 "but these privileges were not granted because GrantSudoPrivileges was not set or set to false.",
                 FailureLevel.ERROR,
             )
+
+
+class SchedulerPluginOsArchitectureValidator(Validator):
+    """Verify that head node architecture and os combination is supported by Scheduler Plugin."""
+
+    def _validate(self, os, architecture: str, supported_x86, supported_arm64):
```

Comment:
are supported_x86 and supported_arm64 initialized to empty lists when not specified in SchedulerPlugin?

Reply:
The default of supported_x86 and supported_arm64 initialized to SUPPORTED_OSES = ["alinux2", "centos7", "ubuntu1804", "ubuntu2004"]
Change the code to
```
        self._register_validator(
            SchedulerPluginOsArchitectureValidator,
            os=self.image.os,
            architecture=self.head_node.architecture,
            supported_x86=get_attr(scheduler_definition, "requirements.supported_distros.x86", default=SUPPORTED_OSES),
            supported_arm64=get_attr(
                scheduler_definition, "requirements.supported_distros.arm64", default=SUPPORTED_OSES
            ),
        )
```
If Requirements is not specified, it will validate if it is in SUPPORTED_OSES